### PR TITLE
addition of nesterov optimizers

### PR DESCRIPTION
**Pull request by xdgarrido**
In this pull request we add Nesterov optimizers based on Adam and Lamb. We called Nadam and 
Nlamb. There are just a small text change to run-pretrainning.py. The biggest addition is in optimizers.py.

### CODE REVIEW

**Comment by c0redumb** [comment]
A few things need to be done:
- Documentation in README (and also the TF2.x conversion)
- The one-file scripts should be updated to TF2.x container (left from previous commits)
- Some sanity tests

I think I will find time to do this next week.

File: optimization.py [hunk 1]
```diff
@@ -318,4 +345,243 @@ def _get_variable_name(self, param_name):
...
+    """Whether to use L2 weight decay for `param_name`."""
+    if not self.weight_decay_rate:
+      return False
+    if self.exclude_from_weight_decay:
+      for r in self.exclude_from_weight_decay:
+        if re.search(r, param_name) is not None:
+          return False
+    return True
+
+  def _get_variable_name(self, param_name):
```
Comment (MikalaiDrabovich) [inline optimization.py:453]: looks like the same def _get_variable_name() repeated four times - maybe move to some base class?

Comment (xdgarrido) [reply optimization.py:453]: The idea here is to preserve the original BERT-AdamOptimizer code. This is why, I did it this way. Is more question of correlation and legibility with the original BERT code. This of course 
does not impact run-time.

**Comment by micmelesse** [comment]
Looks good to me.

**Pull request closed without merging**